src/components/WalletConnectModal.tsx

```diff
@@ -0,0 +1,78 @@
+import { FC, MouseEvent } from "react";
+import { AiOutlineClose } from 'react-icons/ai';
+import { useAccount, useConnect } from "wagmi";
+
+interface WalletConnectModalProps {
+  isDisplayModal: boolean
+  onClose: () => void
+}
+
+const WalletConnectModal: FC<WalletConnectModalProps> = ({ isDisplayModal, onClose }) => {
+  const { connect, connectors, isLoading, pendingConnector } = useConnect()
+  const { isConnected } = useAccount()
+  const metamaskConnector = connectors[0]
+  const walletConnectConnector = connectors[1]
+
+  const handleOnClose = (e: MouseEvent<HTMLInputElement>) => {
+    const target = e.target as HTMLElement
+    if (target.id === 'container') {
+      onClose()
+    }
+  }
+
+  if (!isDisplayModal || isConnected) return null;
+
+  return (
+    <div
```

Comment:
I would be doing @piersss a disservice if I didn't ask you to consider using the `dialog` element instead. 

It's beneficial because you get quite a lot for free, including handling focus and pressing escape to close.

Here's a [relatively short overview video](https://www.youtube.com/watch?v=ywtkJkxJsdg).